frontend/src/metabase/nav/components/WhatsNewNotification/WhatsNewNotification.unit.spec.tsx

```diff
@@ -0,0 +1,154 @@
+import type { VersionInfoRecord } from "metabase-types/api";
+import {
+  createMockVersion,
+  createMockVersionInfo,
+  createMockVersionInfoRecord as mockVersion,
+} from "metabase-types/api/mocks";
+import {
+  createMockSettingsState,
+  createMockState,
+} from "metabase-types/store/mocks";
+import { renderWithProviders, screen } from "__support__/ui";
+import * as domUtils from "metabase/lib/dom";
+import { WhatsNewNotification } from "./WhatsNewNotification";
+
+const notification = () => screen.queryByText("See what's new");
+
+const setup = ({
+  isEmbedded = false,
+  lastAcknowledged = null,
+  currentVersion = "v0.48.0",
+  versions = [
+    mockVersion({ version: "v0.48.2" }),
+    mockVersion({ version: "v0.48.1" }),
+    mockVersion({
+      version: "v0.48.0",
+      announcement_url: "https://metabase.com/releases/48",
+    }),
+    mockVersion({ version: "v0.47.0" }),
+  ],
+}: {
+  isEmbedded?: boolean;
+  currentVersion?: string;
+  lastAcknowledged?: string | null;
+  versions?: VersionInfoRecord[];
+}) => {
+  jest.spyOn(domUtils, "isWithinIframe").mockReturnValue(isEmbedded);
+
+  const versionMock = createMockVersion({ tag: currentVersion });
+
+  const [latest, ...older] = versions;
+  const mockSettings = createMockSettingsState({
+    version: versionMock,
+    "version-info": createMockVersionInfo({ latest, older }),
+    "last-acknowledged-version": lastAcknowledged,
+  });
+
+  return renderWithProviders(<WhatsNewNotification></WhatsNewNotification>, {
+    storeInitialState: createMockState({ settings: mockSettings }),
+  });
+};
+
+describe("WhatsNewNotification", () => {
+  describe("display logic", () => {
```

Comment:
Most of this tests could be moved to the utility function because there is no code in the component that does this